### PR TITLE
doc: add upgrade guide for 0.15+

### PR DESCRIPTION
**Pull request by camilamacedo86**
doc: add upgrade guide for 0.15+

### CODE REVIEW

File: doc/migration/version-upgrade-guide.md [hunk 1]
```diff
@@ -756,6 +756,60 @@ The Helm operator packages and base image were upgraded from Helm v2 to Helm v3.
...
+replace github.com/openshift/api => github.com/openshift/api v0.0.0-20190924102528-32369d4db2ad // Required until https://github.com/operator-framework/operator-lifecycle-manager/pull/1241 is resolved
+```
+
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
+- Run the command `operator-sdk generate crds` to regenerate CRDs
+
+### Breaking Changes on Commands 
+
+This release contains break changes in the commands tool. Note that:
```
Comment (jmccormick2001) [inline doc/migration/version-upgrade-guide.md]: suggest:

This release contains breaking changes in some commands.

File: doc/migration/version-upgrade-guide.md [hunk 2]
```diff
@@ -756,6 +756,60 @@ The Helm operator packages and base image were upgraded from Helm v2 to Helm v3.
...
+
+- The `operator-sdk olm-catalog gen-csv` was replaced by `operator-sdk generate csv`
+- The `operator-sdk up local` is now `operator-sdk run --local`. However, all functionality of this is the same.
```
Comment (jmccormick2001) [inline doc/migration/version-upgrade-guide.md]: suggest:

However, all functionality of this command is retained.

File: doc/migration/version-upgrade-guide.md [hunk 3]
```diff
@@ -756,6 +756,60 @@ The Helm operator packages and base image were upgraded from Helm v2 to Helm v3.
...
+- And then, the `operator-sdk alpha olm [sub-commands] [flags]` was moved from `alpha` to its own sub-command. However, all functionality of this still the same. To check run; `operator-sdk olm --help`.
```
Comment (jmccormick2001) [inline doc/migration/version-upgrade-guide.md]: suggest:

However, all functionality of this command is still the same.

File: doc/migration/version-upgrade-guide.md [hunk 4]
```diff
@@ -756,6 +756,60 @@ The Helm operator packages and base image were upgraded from Helm v2 to Helm v3.
...
+- **(Useful for Ansible and Helm based-operators only)** The `operator-sdk run ansible/helm` are now hidden commands in `exec-entrypoint ansible/helm`. However, all functionality of each sub-command still the same.
```
Comment (jmccormick2001) [inline doc/migration/version-upgrade-guide.md]: suggest:

However, all functionality of each sub-command is still the same.

**Comment by camilamacedo86** [comment]
Hi @jmccormick2001, 

Tks for the help. All suggestions are applied. Please, let me know wdyt?

**Pull request merged**